media-video/mpv/mpv-0.11.0-r1.ebuild

```diff
@@ -0,0 +1,278 @@
+		luajit? ( dev-lang/luajit:2 )
+	)
+	openal? ( >=media-libs/openal-1.13 )
+	pulseaudio? ( media-sound/pulseaudio )
+	rubberband? ( >=media-libs/rubberband-1.8.0 )
+	samba? ( net-fs/samba )
+	sdl? ( media-libs/libsdl2[sound,threads,video,X?,wayland?] )
+	v4l? ( media-libs/libv4l )
+	wayland? (
+		>=dev-libs/wayland-1.6.0
+		>=x11-libs/libxkbcommon-0.3.0
+	)
+	X? (
+		x11-libs/libX11
+		x11-libs/libXext
+		>=x11-libs/libXrandr-1.2.0
+		opengl? (
+			x11-libs/libXdamage
+			virtual/opengl
+		)
+		vaapi? ( >=x11-libs/libva-1.2.0[X] )
+		vdpau? ( >=x11-libs/libvdpau-0.2 )
+		xinerama? ( x11-libs/libXinerama )
+		xscreensaver? ( x11-libs/libXScrnSaver )
+		xv? ( x11-libs/libXv )
+	)
+"
+DEPEND="${CDEPEND}
+	${PYTHON_DEPS}
+	>=dev-lang/perl-5.8
+	dev-python/docutils
+	virtual/pkgconfig
+	doc? ( dev-python/rst2pdf )
+	test? ( >=dev-util/cmocka-1.0.0 )
+"
+RDEPEND="${CDEPEND}
+	selinux? ( sec-policy/selinux-mplayer )
+"
+
+pkg_pretend() {
+	if ! use libass; then
+		ewarn "You have disabled the libass flag. No OSD or subtitles will be displayed."
+	fi
+
+	if use openal; then
+		ewarn "You have enabled the openal audio output, which is a fallback"
+		ewarn "and disabled by upstream."
+	fi
+
+	if use sdl; then
+		ewarn "You have enabled the sdl video and audio outputs, which are fallbacks"
+		ewarn "and disabled by upstream."
+	fi
+
+	if use libav; then
+		einfo "You have enabled media-video/libav instead of media-video/ffmpeg."
+		einfo "Upstream recommends media-video/ffmpeg, as some functionality is not"
+		einfo "provided by media-video/libav."
+	fi
+
+	einfo "mpv optionally supports many different audio and video formats."
+	einfo "You will need to enable support for the desired formats in your"
+	einfo "libavcodec/libavformat provider:"
+	einfo "    media-video/ffmpeg or media-video/libav"
+}
+
+src_prepare() {
+	cp "${DISTDIR}/waf-${WAF_PV}" "${S}"/waf || die
+	chmod 0755 "${S}"/waf || die
+
+	epatch "${FILESDIR}/${PN}-fix-include-in-tests.patch"
+	epatch_user
+}
+
+src_configure() {
+	local mywafargs=(
+		--confdir="${EPREFIX}"/etc/${PN}
+		--docdir="${EPREFIX}"/usr/share/doc/${PF}
+
+		$(usex cli '' '--disable-cplayer')
+		$(use_enable libmpv libmpv-shared)
+
+		--disable-libmpv-static
+		--disable-static-build
+		--disable-build-date	# Create reproducible build
+		--disable-optimize		# Do not add '-O2' to CFLAGS
+		--disable-debug-build	# Do not add '-g' to CFLAGS
+
+		$(use_enable doc pdf-build)
+		$(use_enable vf-dlopen vf-dlopen-filters)
+		$(use_enable cli zsh-comp)
+		$(use_enable test)
+
+		$(use_enable iconv)
+		$(use_enable samba libsmbclient)
+		$(use_enable lua)
+		$(usex luajit '--lua=luajit' '')
+		$(use_enable libass)
+		$(use_enable libass libass-osd)
+		$(use_enable encode encoding)
+		$(use_enable bluray libbluray)
+		$(use_enable dvd dvdread)
+		$(use_enable dvd dvdnav)
+		$(use_enable cdda)
+		$(use_enable enca)
+		$(use_enable libguess)
+		--disable-uchardet		# Disable uchardet until #569114 is fixed
```

Comment:
A few of us have discussed "Second alternative" and despite being the usual work flow, all agree it is the most painful and troublesome to all concerned. We're talking about an IUSE uchardet which does not yet exist so this obliges a two step process:
1. Create the IUSE uchardet in metadata of mpv, then
2. start masking and or unmasking it under profiles
and it has always been a given that editing under profiles is done by a supporting dev.
Whether it's me or mgorny does not matter as long as it's done